src/BookNavigator/volumes/volumes-provider.js

```diff
@@ -17,88 +16,73 @@ export default class VolumesProvider {
-    /** @type {import('../../BookReader').default} */
-    this.bookreader = bookreader;
-
-    this.component.subPrefix = bookreader.options.subPrefix || "";
-    this.component.hostUrl = baseHost;
-    this.component.viewableFiles = this.viewableFiles;
-
     this.id = "volumes";
     this.label = `Viewable files (${this.volumeCount})`;
-    this.icon = html`${volumesIcon}`;
+    this.icon = html`${viewableFilesIcon}`;
+    this.sortOrderBy = sortTypes.default;
 
-    this.sortOrderBy = sortType.default;
+    this.component = document.createElement("iaux-viewable-files");
+    this.component.addSortToUrl = true;
+    this.component.subPrefix = bookreader.options.subPrefix || "";
+    this.component.baseHost = baseHost;
+    this.component.fileList = [...this.viewableFiles];
+
+    this.sortFilesComponent = document.createElement("iaux-sort-viewable-files");
+    this.sortFilesComponent.fileListRaw = this.viewableFiles;
+    this.sortFilesComponent.addEventListener('fileListSorted', (e) => this.handleFileListSorted(e));
+    this.actionButton = this.sortFilesComponent;
 
     // get sort state from query param
     if (this.bookreader.urlPlugin) {
       this.bookreader.urlPlugin.pullFromAddressBar();
 
       const urlSortValue = this.bookreader.urlPlugin.getUrlParam('sort');
-      if (urlSortValue === sortType.title_asc || urlSortValue === sortType.title_desc) {
+      if (urlSortValue === sortTypes.title_asc || urlSortValue === sortTypes.title_desc) {
         this.sortOrderBy = urlSortValue;
       }
     }
-    this.sortVolumes(this.sortOrderBy);
-  }
 
-  get sortButton() {
-    const sortIcons = {
-      default: html`
-        <button class="sort-by neutral-icon" aria-label="Sort volumes in initial order" @click=${() => this.sortVolumes("title_asc")}>${sortNeutralIcon}</button>
-      `,
-      title_asc: html`
-        <button class="sort-by asc-icon" aria-label="Sort volumes in ascending order" @click=${() => this.sortVolumes("title_desc")}>${sortAscIcon}</button>
-      `,
-      title_desc: html`
-        <button class="sort-by desc-icon" aria-label="Sort volumes in descending order" @click=${() => this.sortVolumes("default")}>${sortDescIcon}</button>
-      `,
-    };
-
-    return sortIcons[this.sortOrderBy];
+    this.sortFilesComponent.sortVolumes(this.sortOrderBy);
+
+    this.onProviderChange(this.bookreader);
   }
 
-  /**
-   * @param {'default' | 'title_asc' | 'title_desc'} sortByType
-   */
-  sortVolumes(sortByType) {
-    let sortedFiles = [];
+  /** @param { SortTypesT } sortType */
+  async handleFileListSorted(event) {
+    const { sortType, sortedFiles } = event.detail;
 
-    const files = this.viewableFiles;
-    sortedFiles = files.sort((a, b) => {
-      if (sortByType === sortType.title_asc) return a.title.localeCompare(b.title);
-      else if (sortByType === sortType.title_desc) return b.title.localeCompare(a.title);
-      else return a.orig_sort - b.orig_sort;
-    });
+    this.viewableFiles = sortedFiles;
```

Comment:
yah. it isn't right now, but i think we should collapse even further. this will take a bit of time to rethink bc it involves item-nav menu config as button ingress. will give rethink without holding pdf theater feature longer than needed. 👍